libs/ledger/src/chain/block.cpp

```diff
@@ -90,11 +90,16 @@ void Block::UpdateDigest()
 
 void Block::UpdateTimestamp()
 {
-  if (body.previous_hash != chain::GENESIS_DIGEST)
+  if (!IsGenesis())
   {
     body.timestamp = GetTime(clock_);
   }
 }
 
+bool Block::IsGenesis() const
+{
+  return body.previous_hash == chain::GENESIS_DIGEST;
```

Comment:
I suspect this check will not work with POS (which will become the default consensus mechanism shortly). Have a chat with @n-hutton about the best way to handle this.

Reply:
This is as applied to current master so once the definition of genesis changes only this method would presumably need to change.

Reply:
Oh yes - this is a great change. It is just at the moment on `master` the behaviour would not be correct if you used the `-pos` flag on the command line.

Actually I think this is highlighting a larger bug. So this is excellent!